scho05-sp2

ondrulin opened the pull request:
Odkaz na moji aplikaci: https://eso.vse.cz/~scho05/sp2/

ondrulin committed:
scho05-sp2

Odkaz na moji aplikaci: https://eso.vse.cz/~scho05/sp2/

ondrulin committed:
Core functionality, excluding ratings and styles

ondrulin committed:
Full functionality with CSS styling

nvbach91 commented:
- chybi indikace nacitani dat ajax spinner
- persistovat i volba serazeni
- po pridani do favorite zobrazit info ze bylo pridano
- zobrazit pocet jokes nasbiranych
- pridat moznost strankovani jokes pomoci parametru `page` v search dotazu
- pridat moznost strankovani ulozenych jokes
- pri zmene dat, ktere se maji ukladat do localStorage, neni vhodne pokazde nacist z localStorage, ale je lepsi si drzet referencni objekt, kterym se bude manipulovat stav a do localStorage pouze zapisovat

ondrulin committed:
Final verze (z obhajoby)

Added pagination, spinner, disabled state for genJoke button, notifications (sweetAlerts), fixed min-height for jokeDisplay, clear favorites functionality
